hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/migration/AccountStateTranslator.java

```diff
@@ -173,12 +177,16 @@ public static Account accountFromOnDiskAccount(@NonNull final OnDiskAccount acco
                 .stakedToMe(account.getStakedToMe())
                 .stakePeriodStart(account.getStakePeriodStart())
                 .firstContractStorageKey(firstContractStorageKey)
-                .headNftId(NftID.newBuilder()
-                        .tokenId(TokenID.newBuilder()
-                                .tokenNum(account.getHeadNftTokenNum())
-                                .realmNum(StaticProperties.getRealm())
-                                .shardNum(StaticProperties.getShard()))
-                        .serialNumber(account.getHeadNftSerialNum()))
+                .headNftId(
```

Comment:
Great catch. 🙌

Created an [issue](https://github.com/hashgraph/hedera-services/issues/12548) for us to review all migrations one more time for this kind of problem.

Reply:
Makes sense. tyvm 🙌